File: go/fixture/go-mini/internal/models/schedule.go
package models

import (
	"errors"
	"fmt"
	"slices"
	"time"
)

// Weekday is a weekday.
type Weekday = time.Weekday

// Schedule is a schedule.
type Schedule struct {
	days []Weekday
}

// NewSchedule creates a new Schedule.
func NewSchedule(days []Weekday) (Schedule, error) {
	if len(days) == 0 {
		return Schedule{}, errors.New("schedule: no days")
	}
	for _, d := range days {
		if d < time.Sunday || d > time.Saturday {
			return Schedule{}, fmt.Errorf("schedule: bad weekday %d", d)
		}
	}
	return Schedule{days: days}, nil
}

// Days returns the days.
func (s Schedule) Days() []Weekday {
	return s.days
}

// Includes returns whether the schedule includes the day.
func (s Schedule) Includes(d Weekday) bool {
	return slices.Contains(s.days, d)
}

File: py/fixture/py-mini/internal/models/schedule.py
"""Weekly schedules."""

# A weekday as datetime.weekday() reports it: 0 is Monday, 6 is Sunday.
Weekday = int


class Schedule:
    """Schedule is a schedule."""

    def __init__(self, days: list[Weekday]) -> None:
        if not days:
            raise ValueError("schedule: no days")
        for d in days:
            if d < 0 or d > 6:
                raise ValueError(f"schedule: bad weekday {d}")
        self._days = days

    def days(self) -> list[Weekday]:
        """Return the days."""
        return self._days

    def includes(self, d: Weekday) -> bool:
        """Return whether the schedule includes the day."""
        return d in self._days
